django/db/backends/oracle/base.py

```diff
@@ -386,10 +386,24 @@ class FormatStylePlaceholderCursor:
     def __init__(self, connection):
         self.cursor = connection.cursor()
         # Necessary to retrieve decimal values without rounding error.
-        self.cursor.numbersAsStrings = True
+        self.cursor.outputtypehandler = self._output_type_handler
         # The default for cx_Oracle < 5.3 is 50.
         self.cursor.arraysize = 100
 
+    @staticmethod
+    def _output_type_handler(cursor, name, defaultType, length, precision, scale):
+        '''
+        Called for each db column that fetch from cursor. Return all numbers as
```

Comment:
"Called for each db column fetched from cursors. Return numbers as strings so that decimal values don't have rounding error."